rust/agama-cli/src/questions.rs

```diff
@@ -0,0 +1,49 @@
+use agama_lib::connection;
+use agama_lib::proxies::Questions1Proxy;
+use anyhow::{Context, Ok};
+use clap::{Args, Subcommand, ValueEnum};
+
+
+#[derive(Subcommand, Debug)]
+pub enum QuestionsCommands {
+    /// Set mode for answering questions.
+    Mode(ModesArgs),
+}
+
+#[derive(Args, Debug)]
+pub struct ModesArgs {
+    #[arg(value_enum)]
+    value: Modes,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+pub enum Modes {
+    Interactive,
+    NonInteractive,
+}
+// TODO when more commands is added, refactor and add it to agama-lib and share a bit of functionality
```

Comment:
Beware that the code which is specific to the CLI should live in `agama-cli`. That's the main intention of this package.

Reply:
well, I mean here code that handles questions, do connection, use Question API and so on. But lets see how future will look like. For rust it is nice that strict separation of public and private, so it is easy to refactor internal stuff.